napari/_vispy/vispy_scale_bar_visual.py

```diff
@@ -48,55 +53,78 @@ def __init__(self, viewer, parent=None, order=0):
             self._on_position_change
         )
         self._viewer.camera.events.zoom.connect(self._on_zoom_change)
+        self._viewer.scale_bar.events.font_size.connect(self._on_text_change)
+        self._viewer.scale_bar.events.unit.connect(self._on_dimension_change)
 
         self._on_visible_change(None)
         self._on_data_change(None)
+        self._on_dimension_change(None)
         self._on_position_change(None)
-
-    def _on_zoom_change(self, event):
+        self._on_dimension_change(None)
+
+    def _on_dimension_change(self, _evt=None):
+        """Update dimension"""
+        self._quantity = UNIT_REG(self._viewer.scale_bar.unit)
+        self._on_zoom_change(None, True)
+
+    def _calculate_best_length(self, px_length: float):
+        """calculate new quantity based on the pixel length of the bar."""
```

Comment:
```suggestion
        """Calculate new quantity based on the pixel length of the bar."""
```

Maybe also adding `Parameters` and `Returns` (if applicable?) ?

Not sure if we have any convention for not doing it on private methods 🤷🏼

Reply:
> Not sure if we have any convention for not doing it on private methods 🤷🏼

We do like to add them for private methods. Coverage is not complete, but it is appreciated